sympy/sets/fancysets.py

```diff
@@ -483,6 +492,138 @@ def _intersect(self, other):
             else:
                 return
 
+    def _union(self, other):
+        """
+
+        This function should only be used internally.
+        See Set._union for docstring
+
+        Used within the :class:`Union` class
+
+        Returns
+        =======
+        self._union(other) returns a new, joined ImageSet if self knows how
+        to join itself with other, otherwise it returns ``None``.
+
+        Examples
+        ========
+
+        >>> from sympy import Lambda
+        >>> from sympy.sets.fancysets import ImageSet
+        >>> from sympy import pprint, Union
+        >>> from sympy.core import Dummy, pi, S, Symbol
+        >>> n = Dummy('n')
+        >>> img1 = ImageSet(Lambda(n, 2*n*pi + 3*pi/4), S.Integers)
+        >>> img2 = ImageSet(Lambda(n, 2*n*pi + 7*pi/4), S.Integers)
+        >>> pprint(img1.union(img2), use_unicode=False)
+                3*pi
+        {n*pi + ---- | n in Integers()}
+                 4
+
+        """
+
+        from sympy.polys import Poly
+        from sympy.simplify import simplify
+        from sympy.polys.polyerrors import PolynomialError
+
+        if other is S.EmptySet:
+            return self
+
+        def val_at(imgset, i):
+            base = imgset.base_set
+            if base == S.Reals:
+                return imgset.lamda(i)
+            else:
+                iterable = iter(base)
```

Comment:
because 'Reals' object is not iterable.

```
In [ ]: iter(S.Reals)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-e167fb9b6c63> in <module>()
----> 1 iter(S.Reals)

TypeError: 'Reals' object is not iterable
```

if I want to get first element (at i =0 ), then I am doing 

```
In [21]: ImageSet(Lambda(n, 2*n*pi + 3*pi/4), S.Reals).lamda(0)
Out[21]: 
3⋅π
───
 4 
```

If `base` is not iterable and `S.Reals`, then this may fail.